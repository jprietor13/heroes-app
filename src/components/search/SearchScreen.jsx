import React, { useMemo } from 'react'
import { useForm } from '../../hooks/useForm'
import { getHeroesByName } from '../../helpers/getHeroesByName'
import { HeroCard } from '../hero/HeroCard'

export const SearchScreen = () => {
  const [ {searchText} , handleInputChange ] = useForm()
  const resultHeroes = useMemo(() => getHeroesByName(searchText), [searchText])

  const handleSubmit = (e) => {
    e.preventDefault();
  }

  return (
    <div>
      <h1>Busquedas</h1>
      <hr />
     <div className="row">
       <div className="col-5">
         <h3>Buscar</h3>
         <hr />
         <div className='form-group'>
          <form onSubmit={handleSubmit}>
            <input 
              type="text" 
              className='form-control'
              placeholder='Nombre del personaje...'
              autoComplete='off'
              name="searchText"
              value={searchText}
              onChange={handleInputChange}  
            />
            <button type="submit" className='btn btn-primary mt-3'>Buscar</button>
          </form>
         </div>
       </div>
       <div className='col-7'>
          {resultHeroes.map(heroes => 
            <HeroCard key={heroes.id} {...heroes}/> 
          )}
       </div>
     </div>
    </div>
  )
}
