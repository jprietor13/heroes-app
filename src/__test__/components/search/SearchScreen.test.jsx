import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { SearchScreen } from "../../../components/search/SearchScreen";

describe("Pruebas en <SearchScreen />", () => {
  test("Debe cargarse correctamente", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchScreen />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert-info").text().trim()).toBe(
      "Ningun heroe a buscar!!"
    );
  });

  test("Debe mostrar a batman y el input con el valor del queryString", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(wrapper.find(".form-control").prop("value")).toBe("batman");
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el mensaje de error si no se encuentra el heroe", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search?q=batman123"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(wrapper.find(".alert-danger").text().trim()).toBe(
      "Sin coincidencias para batman123"
    );
  });
});
