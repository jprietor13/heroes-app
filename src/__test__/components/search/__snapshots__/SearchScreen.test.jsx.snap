// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> Debe cargarse correctamente 1`] = `
<div>
  <h1>
    Busquedas
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h3>
        Buscar
      </h3>
      <hr />
      <div
        className="form-group"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchText"
            onChange={[Function]}
            placeholder="Nombre del personaje..."
            type="text"
            value=""
          />
          <button
            className="btn btn-primary mt-3"
            type="submit"
          >
            Buscar
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-7"
    >
      <h3>
        Resultados
      </h3>
      <hr />
      <p
        className="alert alert-info"
      >
        Ningun heroe a buscar!!
      </p>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchScreen /> Debe mostrar a batman y el input con el valor del queryString 1`] = `
<div>
  <h1>
    Busquedas
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h3>
        Buscar
      </h3>
      <hr />
      <div
        className="form-group"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchText"
            onChange={[Function]}
            placeholder="Nombre del personaje..."
            type="text"
            value="batman"
          />
          <button
            className="btn btn-primary mt-3"
            type="submit"
          >
            Buscar
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-7"
    >
      <h3>
        Resultados
      </h3>
      <hr />
      <div
        className="col"
      >
        <div
          className="card"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-4"
            >
              <img
                alt="Batman"
                className="card-img-top"
                src="/assets/images/dc-batman.jpg"
              />
            </div>
            <div
              className="col-8"
            >
              <h4
                className="card-title mt-2"
              >
                Batman
              </h4>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                Detective Comics #27
              </p>
              <a
                className=""
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
